Sources/ConcurrencyHelpers/YieldWithBackPressure.swift

```diff
@@ -28,6 +29,9 @@ public func yieldWithBackPressure<Message>(message: Message,
             return true
```

Comment:
IMHO we should remove `discarableResult` here, because it is important for the caller to handle the result properly, and stop doing something if result is 'false'. As of now we have few places where result is not checked and producer will continue yielding messages.

Reply:
Or we can make this function throwable and use `Task.checkCancellation()`.